hudi-utilities/src/main/java/org/apache/hudi/utilities/deltastreamer/SourceFormatAdapter.java

```diff
@@ -123,4 +127,15 @@ public InputBatch<Dataset<Row>> fetchNewDataInRowFormat(Option<String> lastCkptS
   public Source getSource() {
     return source;
   }
+
+  @Override
+  public void close() {
+    if (source instanceof Closeable) {
```

Comment:
Can you create a follow up ticket to implement close for other sources (for eg, for S3Eventssource, we can close/shutdown sqs client).

Reply:
Good call. Created HUDI-4624